lib/Parser/Parse.cpp

```diff
@@ -915,10 +916,21 @@ Symbol* Parser::AddDeclForPid(ParseNodePtr pnode, IdentPtr pid, SymbolType symbo
                 Error(ERRRedeclaration);
             }
         }
+        else if (!pnodeFnc->sxFnc.IsBodyAndParamScopeMerged()
+            && scope->GetScopeType() == ScopeType_FunctionBody
+            && (pnode->nop == knopLetDecl || pnode->nop == knopConstDecl))
+        {
+            // In case of split scope function when we add a new let or const declaration to the body
+            // we have to check whether the param scope already has the same symbol defined.
+            PidRefStack *paramScopeRef = pid->GetPidRefForScopeId(pnodeFnc->sxFnc.pnodeScopes->sxBlock.blockId);
+            if (paramScopeRef && paramScopeRef->GetSym() && !paramScopeRef->GetSym()->GetIsArguments())
```

Comment:
>paramScopeRef->GetSym() [](start = 33, length = 23)

if the sym is null then why GetPidRefForScopeId will return paramScopeRef at all?